source/adios2/toolkit/derived/Function.cpp

```diff
@@ -126,6 +152,87 @@ DerivedData SubtractFunc(std::vector<DerivedData> inputData, DataType type)
+            inputData.begin() + 1, inputData.end(), dataSize, [](T a, T b) { return a * b; },      \
+            (T)1);                                                                                 \
+        for (size_t i = 0; i < dataSize; i++)                                                      \
+            multValues[i] = *(reinterpret_cast<T *>(inputData[0].Data) + i) * multValues[i];       \
+        return DerivedData({(void *)multValues, inputData[0].Start, inputData[0].Count});          \
+    }
+    ADIOS2_FOREACH_PRIMITIVE_STDTYPE_1ARG(declare_type_mult)
+    helper::Throw<std::invalid_argument>("Derived", "Function", "MultFunc",
+                                         "Invalid variable types");
+    return DerivedData();
+}
+
+DerivedData DivFunc(std::vector<DerivedData> inputData, DataType type)
+{
+    PERFSTUBS_SCOPED_TIMER("derived::Function::DivFunc");
+    size_t dataSize = std::accumulate(std::begin(inputData[0].Count), std::end(inputData[0].Count),
+                                      1, std::multiplies<size_t>());
+
+#define declare_type_div(T)                                                                        \
+    if (type == helper::GetDataType<T>())                                                          \
+    {                                                                                              \
+        T *divValues = detail::ApplyOneToOne<T>(                                                   \
+            inputData.begin() + 1, inputData.end(), dataSize, [](T a, T b) { return a / b; },      \
+            (T)1);                                                                                 \
+        for (size_t i = 0; i < dataSize; i++)                                                      \
+            divValues[i] = *(reinterpret_cast<T *>(inputData[0].Data) + i) * divValues[i];         \
+        return DerivedData({(void *)divValues, inputData[0].Start, inputData[0].Count});           \
+    }
+    ADIOS2_FOREACH_PRIMITIVE_STDTYPE_1ARG(declare_type_div)
+    helper::Throw<std::invalid_argument>("Derived", "Function", "DivFunc",
+                                         "Invalid variable types");
+    return DerivedData();
+}
+
+DerivedData SqrtFunc(std::vector<DerivedData> inputData, DataType type)
+{
+    PERFSTUBS_SCOPED_TIMER("derived::Function::SqrtFunc");
+    size_t dataSize = std::accumulate(std::begin(inputData[0].Count), std::end(inputData[0].Count),
+                                      1, std::multiplies<size_t>());
+
+#define declare_type_sqrt(T)                                                                       \
+    if (type == helper::GetDataType<T>())                                                          \
+    {                                                                                              \
+        T *sqrtValues = detail::ApplySqrt<T>((T *)inputData[0].Data, dataSize);                    \
```

Comment:
We could use ApplyOneToOne, set the data to all zeros, then define the computation as adding the square root of the data to the zeroed term, but it seemed silly to set the data to 0 just to add the value of interest, especially since we should only be calling sqrt on a single DerivedData expression. I created a ApplyOneToOneOnce function in the TrigFns pr for operators that should only have one input - I could pull that in here and use that instead.

Reply:
But as for pow, I'm not sure how we could use pow with ApplyOneToOne because we need both the base and the exponent to calculate pow

Reply:
I'll look at the new version but I think your new function should work